src/pykeen/evaluation/evaluator.py

```diff
@@ -449,12 +469,31 @@ def evaluate(
         Return a single instance of :class:`MetricResults` if only one evaluator was given.
     :param use_tqdm:
         Should a progress bar be displayed?
+    :param restrict_entities_to:
+        Optionally restrict the evaluation to the given entity IDs. This may be useful if one is only interested in a
+        part of the entities, e.g. due to type constraints, but wants to train on all available data. For ranking the
+        entities, we still compute all scores for all possible replacement entities to avoid irregular access patterns
+        which might decrease performance, but the scores with afterwards be filtered to only keep those of interest.
+        If provided, we assume that the triples are already filtered, such that it only contains the entities of
+        interest.
+    :param do_time_consuming_checks:
+        Whether to perform some time consuming checks on the provided arguments. Currently, this encompasses:
+        - If restrict_entities_to is not None, check whether the triples have been filtered.
+        Disabling this option can accelerate the method.
     """
     if isinstance(evaluators, Evaluator):  # upgrade a single evaluator to a list
         evaluators = [evaluators]
 
     start = timeit.default_timer()
 
+    # verify that the triples have been filtered
+    if restrict_entities_to is not None and do_time_consuming_checks:
+        present_entity_ids = set(mapped_triples[:, 0].unique().tolist()).union(mapped_triples[:, 2].unique().tolist())
```

Comment:
do we have a utility somewhere that gets the entity list from a set of triples? We've rewritten this line so many times

Reply:
6e294b3